CMakeLists.txt

```diff
@@ -521,14 +521,25 @@ endif()
 if(runtime_cxxmodules)
   ROOT_GET_LIBRARY_OUTPUT_DIR(library_output_dir)
   get_property(modules_idx_deps GLOBAL PROPERTY modules_idx_deps_property)
-  add_custom_command(OUTPUT ${library_output_dir}/modules.idx
-                     COMMAND ${CMAKE_COMMAND} -E remove -f modules.idx modules.timestamp
-                     COMMAND ${ld_library_path}=${library_output_dir}:$ENV{${ld_library_path}}
-                             ROOTIGNOREPREFIX=1 ROOT_HIST=0
-                             $<TARGET_FILE:root.exe> -l -q -b
-                     WORKING_DIRECTORY ${library_output_dir}
-                     DEPENDS $<TARGET_FILE:root.exe> Cling Hist Tree Gpad Graf HistPainter move_artifacts
-                             ${modules_idx_deps})
+  if(WIN32)
+    add_custom_command(OUTPUT ${library_output_dir}/modules.idx
+                       COMMAND ${CMAKE_COMMAND} -E remove -f modules.idx modules.timestamp
+                       COMMAND set PATH=${CMAKE_RUNTIME_OUTPUT_DIRECTORY} &&
```

Comment:
Why `ROOT_GET_LIBRARY_OUTPUT_DIR` does not get the appropriate location here? We probably need to fix that to avoid the branchy code which is hard to maintain.

Reply:
What do you mean? This is the right location, i.e. the bin directory, where the DLLs are

Reply:
Yes, but why the ROOT_GET_LIBRARY_OUTPUT_DIR  function we call did not return the right path and we need to adjust it conditionally?

Reply:
Ping on this comment.

Reply:
Fixed